app/pods/publications/publication/documents/controller.js

```diff
@@ -384,64 +255,46 @@ export default class PublicationDocumentsController extends Controller {
 
   @action
   async onPerformFilter(filter) {
-    this.renderPieces = false;
-    this.selectedPieces = [];
-    this.filter = filter;
-    await this.sortAndFilterPieces();
-    this.renderPieces = true;
+    FilterQueryParams.updateFromFilterAndReload(this, filter);
   }
 
   async sortAndFilterPieces() {
     this.showLoader = true;
-    const pieces = this.model.case.sortedPieces;
-    this.filteredSortedPieces = [];
-    for (let index = 0; index < pieces.length; index++) {
-      const piece = pieces[index];
-      // sync filter first
-      if (!this.filterTitle(piece)) {
-        continue;
-      }
-      if (!await this.filterFileType(piece)) {
-        continue;
-      }
-      if (!await this.filterPieceType(piece)) {
-        continue;
+    const sortedPieces = sortPieces(this.model);
+
+    // Als we geen types hebben geselecteerd, laten we alles zien.
+    if (!this.isFilterFileTypeActive()) {
+      this.filteredSortedPieces = sortedPieces;
+    } else {
+      // Filtering of file extensions is not yet possible in the backend, so we do it here.
+      const filteredSortedPieces = [];
+      for (const piece of sortedPieces) {
+        if (!await this.filterFileType(piece)) {
+          continue;
+        }
+        filteredSortedPieces.push(piece);
       }
-      this.filteredSortedPieces.pushObject(piece);
+
+      this.filteredSortedPieces = filteredSortedPieces;
     }
+
     this.showLoader = false;
   }
 
-  filterTitle(piece) {
-    return piece.name.toLowerCase().includes(this.filter.documentName.toLowerCase());
+  isFilterFileTypeActive() {
+    return !!this.filter.fileTypes.length;
   }
 
   async filterFileType(piece) {
-    // Als we geen types hebben geselecteerd, laten we alles zien.
-    if (this.filter.fileTypes.length === 0) {
-      return true;
-    }
-
-    const ext = await piece.get('file.extension');
+    // await since not "include"-ed in query
+    const file = await piece.get('file');
+    const ext = file.extension;
```

Comment:
Ik had dit eerst niet awaited. Dit gaf een fout bij refresh met file-extension-filter geselecteerd. Ik ben er nog niet uit wanneer ge await piece.get('file') moet gebruiken en wanneer `await piece.file` volstaat.

Reply:
Deze is op zich logisch, bij laden van pieces zit er geen includes van de file.
Als je net hebt geupload, dan werkt dat wel omdat piece en file al in de lokale storage zitten